api/fsi_test.go

```diff
@@ -53,3 +59,131 @@ func TestFSIHandlers(t *testing.T) {
 	}
 	runHandlerTestCases(t, "checkout", h.CheckoutHandler(""), checkoutCases, true)
 }
+
+func TestNoHistory(t *testing.T) {
+	node, teardown := newTestNode(t)
+	defer teardown()
+
+	inst := newTestInstanceWithProfileFromNode(node)
+
+	initDir := "fsi_tests/init_dir"
+	if err := os.MkdirAll(initDir, os.ModePerm); err != nil {
+		panic(err)
+	}
+	defer os.RemoveAll(filepath.Join("fsi_tests"))
+
+	// Create a linked dataset without saving, it has no versions in the repository
+	f := fsi.NewFSI(node.Repo)
+	ref, err := f.InitDataset(fsi.InitParams{
+		Filepath: initDir,
+		Name:     "test_ds",
+		Format:   "csv",
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if ref != "peer/test_ds" {
+		t.Errorf("expected ref to be \"peer/test_ds\", got \"%s\"", ref)
+	}
+
+	dsHandler := NewDatasetHandlers(node, false)
+
+	// Dataset with no history
+	actualStatusCode, actualBody := APICall("/peer/test_ds", dsHandler.GetHandler)
+	if actualStatusCode != 422 {
+		t.Errorf("expected status code 422, got %d", actualStatusCode)
+	}
+	expectBody := `{ "meta": { "code": 422, "error": "no history" }, "data": null }`
+	if expectBody != actualBody {
```

Comment:
these expect vs actual comparisons would benefit from that new go-cmp package 'Diff' method for easier test maintenance. Not a vital change tho